### ISSUE TITLE
Replacing locale at runtime

### ISSUE THREAD

**Issue by TheLeonsver1** [kgv/bevy_fluent#4]
I've tried to look at the code for ways to replace the chosen `locale` at runtime and I've noticed
that `Snapshot` has an `IndexMap` for the locales and their bundles.

But `Snapshot`'s only being loaded with one locale(and I've discussed this with the community and they agree that it's better to hold in memory only the bundle for the locale used), the `default` locale that's set inside `FluentSettings`.

(It should be mentioned that I haven't checked if you load all the locale assets and keep them in memory even though they are not loaded into `Snapshot`)

I think that in this case, the `IndexMap` can be replaced with just a struct that holds a `bundle` and the `locale` used.

I've also seen that you cannot replace the locale used in-game unless you `init_resource` for the `Snapshot` again(after replacing the locale in `FluentSettings`), but that is not possible at the current version because you check that the resource wasn't already initiated in the world.

So it's currently impossible to replace the locale at runtime which I think would be really useful for your users.
would love to hear your input about these issues, thanks for the crate!

**Comment by kgv** [comment]
I am still undecided on the architecture of the plugin.

Snapshot from the `main` branch contains `locale fallback chain` only for current settings (not all available locales).
[You can see how it constructs](https://docs.rs/fluent-locale/0.10.1/fluent_locale/negotiate/index.html#the-exact-algorithm-is-custom-and-consists-of-a-6-level-strategy)

You can take a look at [`debug` branch](https://github.com/kgv/bevy_fluent/tree/debug). This branch implements locale change.

**Comment by TheLeonsver1** [comment]
Great :D would you be fine with adding this crate to 'bevy_awesome`?
I've already opened an issue about adding it to a new category there:
https://github.com/bevyengine/awesome-bevy/issues/96

**Comment by kgv** [comment]
Thank you for your help.

I was planning to announce this plugin (including adding it to awesome-bevy) as soon as I have precisely defined its internal architecture.
But I think the current `debug` branch is what I wanted.
I think it will be quite possible to announce when the `debug` branch will be merged to `main` branch.

**Comment by TheLeonsver1** [comment]
Cool! glad to hear :D

**Comment by TheLeonsver1** [comment]
Hi, i'm trying to use the `debug` branch and ran into a problem with the `syn` dependency, 
I fixed it with 
```
[patch.'https://github.com/kgv/format']
format = { version = "0.2.4", git = "https://github.com//kgv/format", branch = "syn-1.0.65", optional = true, features = ["ext"] }
```
with help from the bevy discord, 
could `bevy_fluent` depend on that branch of `format` by default maybe? at least for the 0.5 release, 

I've been told that `syn` was updated on `main` already but bevy_fluent's `debug` branch is for `0.5` right?

**Comment by TheLeonsver1** [comment]
I think there might be another issue, the assets loaded by `FluentServer` are never registered as `LoadState::Loaded` it seems. That is sort of an issue since I can't know when I can use them to update my text this way.

**Comment by kgv** [comment]
`branch = "syn-1.0.65"` - the correct branch, it is my fail that i didn't change it in `debug`.

> I've been told that syn was updated on main already but bevy_fluent's debug branch is for 0.5 right?

right

Be careful, the `debug` branch is not intended to be used, it can change at any time.

I am waiting bevyengine/bevy#2224 for merge `debug` with `main`.

> assets loaded by FluentServer are never registered as LoadState::Loaded

The resulting asset (`Localization`) is not actually loaded. It is created based on several loaded components.
For this reason, it uses `HandleId::Id`, not `HandleId::AssetPathId`.
And (from the `bevy` source) `HandleId::Id` will always have the status `LoadState::NotLoaded`. 

```rust
match handle.into() {
    HandleId::AssetPathId(id) => {...}
    HandleId::Id(_, _) => LoadState::NotLoaded,
}
```

In the example, I solved this problem like [this](https://github.com/kgv/bevy_fluent/blob/e861ade80e8b12caf48740605d64c7c24e6f94d0/examples/ui/main.rs#L50).

**Comment by TheLeonsver1** [comment]
oh, got it, thank you very much for the help! :)

**Comment by kgv** [comment]
Fixed: #5
